### PR TITLE
add async/await to runtime plugins

### PR DESCRIPTION
**Pull request by Nikola-3**
This is the first in a series of PRs that move over atlaspack rust plugins to async/await syntax to make them run asynchronously. The PRs are separated by plugin type.

### CODE REVIEW

File: crates/atlaspack_core/Cargo.toml [hunk 1]
```diff
@@ -28,3 +28,4 @@ serde-value = { workspace = true }
 tracing = { workspace = true }
 tracing-subscriber = { workspace = true }
 xxhash-rust = { workspace = true, features = ["xxh3"] }
+tokio = "1.41.0"
```
Comment (alshdavid) [inline crates/atlaspack_core/Cargo.toml:31]: Try this to pull the version specified in the root `Cargo.toml`
```toml
tokio = { workspace = true }
```

Comment (alshdavid) [reply crates/atlaspack_core/Cargo.toml:31]: Actually, `tokio` isn't used in this crate so this can be removed

Comment (Nikola-3) [reply crates/atlaspack_core/Cargo.toml:31]: Oh right that was a miss from the test auto complete 👍

File: crates/atlaspack_core/src/plugin/runtime_plugin.rs [hunk 2]
```diff
@@ -21,9 +22,10 @@ pub struct RuntimeAsset {
 }
 
 /// Programmatically insert assets into bundles
+#[async_trait]
 pub trait RuntimePlugin: Debug {
```
Comment (alshdavid) [inline crates/atlaspack_core/src/plugin/runtime_plugin.rs:26]: In Rust, a type needs to be marked as "safe" to be sent across threads. to do this you need to add the `Send` and sometimes `Sync` type bounds.

The reasons async types are not constrained as thread safe by default is Rust allows developers to spawn Futures on the current thread (does not need to be thread safe) or multiple threads (needs to be thread safe).

```rust
pub trait RuntimePlugin: Debug + Send + Sync {}
```

Comment (Nikola-3) [reply crates/atlaspack_core/src/plugin/runtime_plugin.rs]: Done

**Pull request merged**